tests/Doctrine/Tests/ORM/Functional/Ticket/GH10868Test.php

```diff
@@ -0,0 +1,192 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Doctrine\Tests\ORM\Functional\Ticket;
+
+use Doctrine\Common\Collections\ArrayCollection;
+use Doctrine\ORM\Mapping as ORM;
+use Doctrine\Tests\OrmFunctionalTestCase;
+
+class GH10868Test extends OrmFunctionalTestCase
+{
+    protected function setUp(): void
+    {
+        parent::setUp();
+
+        $this->setUpEntitySchema([
+            GH10868Shop::class,
+            GH10868Offer::class,
+            GH10868Order::class,
+            GH10868OrderProduct::class,
+        ]);
+    }
+
+    public function testReferenceAndLazyLoadProxyAreTheSame(): void
+    {
+        $shop         = new GH10868Shop();
+        $offer        = new GH10868Offer($shop, 1);
+        $order        = new GH10868Order();
+        $orderProduct = new GH10868OrderProduct();
+
+        $orderProduct->order        = $order;
+        $orderProduct->productOffer = $offer;
+        $order->shop                = $shop;
+        $order->orderProducts->add($orderProduct);
+
+        $this->_em->persist($shop);
+        $this->_em->persist($offer);
+        $this->_em->persist($order);
+        $this->_em->persist($orderProduct);
+        $this->_em->flush();
+        $this->_em->clear();
+
+        $order->orderProducts->count();
```

Comment:
We should not do that, since `$order` belongs to an EM instance that has been cleared.